internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -0,0 +1,397 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package kubernetessecret
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	secretmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+	secretexportmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster/secretexport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kubernetessecret/scope"
+)
+
+func ResourceSecret() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   dataSourceSecretRead,
+		CreateContext: resourceSecretCreate,
+		DeleteContext: resourceSecretDelete,
+		Schema:        secretSchema,
+	}
+}
+
+var secretSchema = map[string]*schema.Schema{
+	NameKey: {
```

Comment:
Add descriptions to all the schema components.

Reply:
added